xknx/management/procedures.py

```diff
@@ -58,3 +66,115 @@ async def nm_individual_address_check(
+    dev_pgm_mode = await nm_individual_address_read(
+        xknx, raise_if_multiple=True
+    )  # raises exception if more than one device in programming mode
+    if len(dev_pgm_mode) == 0:
+        logger.debug("No device in programming mode detected.")
+        raise ManagementConnectionError("No device in programming mode detected.")
+
+    # check if new and received addresses match
+    if address_found:
+        if individual_address != dev_pgm_mode[0]:
+            logger.debug(
+                "Device with address %s found and it is not in programming mode. Exiting to prevent address conflict.",
+                individual_address,
+            )
+            raise ManagementConnectionError(
+                f"A device was found with {individual_address}, cannot continue with programming."
+            )
+        # device in programming mode's address matches address that we want to write, so we can abort the operation safely
+        logger.debug("Device already has requested address, no write operation needed.")
+    else:
+        await xknx.management.send_broadcast(
+            Telegram(
+                GroupAddress("0/0/0"),
+                payload=apci.IndividualAddressWrite(address=individual_address),
+            )
+        )
+        logger.debug("Wrote new address %s to device.", individual_address)
+
+    async with xknx.management.connection(
+        address=IndividualAddress(individual_address)
+    ) as connection:
+        logger.debug(
+            "Checking if device exists at %s and restarting it.", individual_address
+        )
+
+        try:
+            response = await connection.request(
+                payload=apci.DeviceDescriptorRead(descriptor=0),
+                expected=apci.DeviceDescriptorResponse,
+            )
+        except ManagementConnectionTimeout as ex:
+            # if nothing is received (-> timeout) IA is free
+            logger.debug("No device answered to connection attempt. %s", ex)
+            response = None
+        if response and isinstance(response.payload, apci.DeviceDescriptorResponse):
+            # if response is received IA is occupied
+            logger.debug("Device found at %s", individual_address)
+        else:
+            raise ManagementConnectionError(
+                f"Failed to detect individual address ({individual_address}) after write address operation."
+            )
+
+        logger.debug("Restating device, exiting programming mode.")
+        # A_Restart will not be ACKed by the device, so it is manually sent to avoid timeout and retry
+        seq_num = next(connection.sequence_number)
+        telegram = Telegram(
+            destination_address=connection.address,
+            source_address=xknx.current_address,
+            payload=apci.Restart(),
+            tpci=tpci.TDataConnected(sequence_number=seq_num),
+        )
+        await xknx.cemi_handler.send_telegram(telegram)
+
+    return True
```

Comment:
```suggestion
```
Since we don't return anything but `True`, we can also omit it and annotate with `-> None:`.